### PR TITLE
Use RRTMG version of  maximum-random overlap algorithm.

### LINKED ISSUE #312: Change the random number generation in maximum overlap assumption
Currently, we are using the [maximum random overlapping assumption from RRTMGP](https://github.com/earth-system-radiation/rte-rrtmgp/blob/5ae05b41dc345dd9cdc0c5a28014df892df3af3b/extensions/cloud_optics/mo_cloud_sampling.F90#L125). We want to change it to the one in RRTMG (e.g. [here](https://github.com/AER-RC/RRTMG_SW/blob/286e84ed14f61e2279ba819f4ce512a30b48f0b3/src/mcica_subcol_gen_sw.f90#L517)). 

The difference between this two is that whether new random numbers are generated depends on whether there is cloud in the entire adjacent layer in rrtmgp and whether there is cloud in the adjacent layer in the subcolumn (g-point) in rrtmg. When new random numbers are generated, they are also multiplied by (1 - cloud fraction of the adjacent layer) in rrtmg, but not in rrtmgp.

### PR DESCRIPTION
**Pull request by sriharshakandala**
# PULL REQUEST

## Purpose and Content
Implement the RRTMG version of the maximum-random overlap algorithm.

## Linked Issues
(Provide references to any link issues. Use closes #issuenum to automatically close an open issue)
- Fixes #312 
- Closes #312 

## PR Checklist
- [x] This PR has a corresponding issue OR is linked to an SDI.
- [x] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [x] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [x] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [x] I linted my code on my local machine prior to submission OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code used in an integration test OR N/A.
- [x] All tests ran successfully on my local machine OR N/A.
- [x] All classes, modules, and function contain docstrings OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by sriharshakandala** [comment]
bors try

File: src/optics/CloudOptics.jl [hunk 1]
```diff
@@ -201,14 +201,19 @@ end
...
 """
 function build_cloud_mask!(cld_mask, cld_frac, random_arr, gcol, ::MaxRandomOverlap)
 
     FT = eltype(cld_frac)
-    local_rand = view(random_arr, :, gcol)
+    local_rand = view(random_arr, :, :, gcol)
     local_cld_frac = view(cld_frac, :, gcol)
     local_cld_mask = view(cld_mask, :, :, gcol)
-    ngpt, nlay = length(local_rand), length(local_cld_frac)
+    ngpt, nlay = size(local_rand, 1), length(local_cld_frac)
```
Comment (szy21) [inline src/optics/CloudOptics.jl:213]:
```suggestion
    ngpt, nlay = size(local_rand, 1), size(local_rand, 2)
```
Would this be slightly better (easier to understand)? I don't have a preference, up to you.

**Review by szy21 — APPROVED** [review]

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Pull request merged**